"Select board and port in Arduino IDE" – improved and updated for 2.0.0 (#128)
arduino/help-center-content: content/Software Support/Upload/Select-board-and-port-in-Arduino-IDE.md

* The **board** selection tells Arduino IDE how to compile the sketch and what protocols to use when uploading the sketch. In some cases, it will be automatically set, but it can be overridden with the _Tools > Board_ menu or in the "Select other board and port..." dialog.
* The **port** selection refers to an actual board (or other device) that's connected to your computer. Many boards are automatically recognized by Arduino IDE—these boards will have their name displayed next to their port in the _Tools > Port_ menu and will have their port associated with the identified board in the board selector menu.

Port and board selection can be managed in two ways:

* [Using the board selector (requires IDE 2)](#selector)
* [Using the Tools menu](#tools-menu)

---

<a id="selector"></a>

## Using the board selector

> **Note:** The board selector is only available in IDE 2.

Follow these steps to use a connected board:

1. Find the board selector and click to open.

2. A list of ports will be displayed. If a board could be identified, the board name will be displayed, otherwise, it will display "Unknown".

   ![The board selector menu in Arduino IDE 2.](img/board-selector-labels.png)

> 3. Click a port to select it:

Reviewer: ```suggestion
3. Click on a port to select it:
```
typo

>    * If a board core was identified, it will automatically be selected. The selected board core will be displayed. The "![Port selected symbol.](img/symbol_port2.png)" indicates that a port is selected.

Author: ```suggestion
   * If a board was identified, it will automatically be selected. The selected board name will be displayed. The "![Port selected symbol.](img/symbol_port2.png)" indicates that a port is selected.
```

   |----------------------------------------------|-----------------------------------------------------------------------------|
   | Arduino AVR Boards                           | Arduino Uno, Arduino Mega, Arduino Nano (classic)                           |
   | Arduino megaAVR Boards                       | Arduino Uno WiFi Rev2, Arduino Nano Every                                   |
   | Arduino SAMD (32-bits ARM Cortex-M0+) Boards | MKR Zero, MKR WiFi 1000, MKR WiFi 1010, and other MKR boards                |
   | Arduino SAM (32-bits ARM Cortex-M3) Boards   | Arduino Due                                                                 |
   | Mbed OS Nano Boards                          | Arduino Nano 33 BLE, Arduino Nano 33 BLE Sense, Arduino Nano RP2040 Connect |
   | Mbed OS Portenta Boards                      | Portenta H7                                                                 |

   > If you don't know which package to use, or if it's missing from the list, see [Add a board to Arduino IDE](https://support.arduino.cc/hc/en-us/articles/360016119519-How-to-add-boards-in-the-board-manager).

> 3. Hover over a package to reveal the board cores.

Reviewer 2: ```suggestion
3. Hover over a package to reveal the boards.
```
